crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1378,7 +1383,15 @@ impl PgManager {
                     Coin::try_from(stored_obj)
                         .map_err(|e| eprintln!("Error converting object to coin: {:?}", e))
                         .ok()
-                        .map(|coin| Edge::new(coin.move_obj.native_object.id().to_string(), coin))
+                        .map(|coin| {
+                            Edge::new(
+                                coin.move_obj
+                                    .native_object
+                                    .id()
+                                    .to_canonical_string(/* with_prefix */ true),
```

Comment:
I'm actually not super sure if we should be using to_canonical_string here and below for the `staked_sui_id` since I don't know what this `Edge` thing does.. let me know if this is incorrect @wlmyng

Reply:
The first segment to Edge::new() is an opaque cursor, which so happens to be the object id rn. I think this need not be the canonical string format since we allow users to put in both shorthand or the canonical format

Reply:
I think it's fine/good to make this canonical for now.  Will is right that the format of this will change to make it opaque, but this output represents a unique identifier coming back from the database, so it should be canonicalised (i.e. even if we make it opaque, we would canonicalise it first).